### PR TITLE
Retry sending email confirmation

### PR DESCRIPTION
**Pull request by xitronix**

### CODE REVIEW

File: universal-login-sdk/test/core/services/WalletService.int.test.ts [hunk 1]
```diff
@@ -153,6 +153,8 @@ describe('INT: WalletService', () => {
       expect(walletService.state).to.deep.include({kind: 'RequestedCreating'});
       await promise;
       await expect(walletService.confirmCode('12345')).to.be.rejectedWith('Error: Invalid code: 12345');
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/test/core/services/WalletService.int.test.ts:155]: Code in the relayer will change

File: universal-login-react/src/ui/commons/SnackBarQueue.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    }
+  }, [showIndex]);
+
+  const setTimeoutForIndex = (index: number) => {
+    const timeout_id = setTimeout(() => setShowIndex(index), snackQueue[index].delay * 1000);
+    return () => clearTimeout(timeout_id);
+  };
+
+  return <>
+    {showIndex !== null ? snackQueue[showIndex].element : null}
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/commons/SnackBarQueue.tsx:28]: showIndex && snackQueue[showIndex].element

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 3]
```diff
@@ -190,12 +195,12 @@ export class WalletService {
   }
 
   setRequested(wallet: RequestedCreatingWallet) {
-    ensure(this.state.kind === 'None', WalletOverridden);
+    ensure(this.state.kind === 'None' || this.state.kind === 'RequestedCreating', WalletOverridden);
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/core/services/WalletService.ts:198]: Unnecessary

File: universal-login-sdk/src/core/services/WalletService.ts [hunk 4]
```diff
@@ -190,12 +195,12 @@ export class WalletService {
...
     this.setState({kind: 'RequestedCreating', wallet});
   }
 
   setRequestedRestoring(wallet: RequestedRestoringWallet) {
-    ensure(this.state.kind === 'None', WalletOverridden);
+    ensure(this.state.kind === 'None' || this.state.kind === 'RequestedRestoring', WalletOverridden);
```
Comment (JustynaBroniszewska) [inline universal-login-sdk/src/core/services/WalletService.ts:203]: Unnecessary

File: universal-login-react/src/ui/Onboarding/RequestConfirmationRetry.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    }
+  };
+
+  switch (retryState) {
+    case 'INITIAL':
+      return null;
+    case 'BE_PATIENT':
+      return <SnackWaiting text={'It can take some time... Be patient...'} />;
+    case 'RETRY':
+      return <SnackWithButton buttonText='Retry' text='No email yet?' onClick={handleRetry} />;
```
Comment (JustynaBroniszewska) [inline universal-login-react/src/ui/Onboarding/RequestConfirmationRetry.tsx:45]: We should add default:  null

Comment (xitronix) [reply universal-login-react/src/ui/Onboarding/RequestConfirmationRetry.tsx:45]: Default should be error if we want default case

**Review by JustynaBroniszewska — APPROVED** [review]

**Pull request merged**